### ISSUE TITLE
Incorrect Float comparison

### ISSUE THREAD

**Issue by Adven27** [lukas-krecan/JsonUnit#468]
Using `net.javacrumbs.json-unit:json-unit:2.31.0`.

I expected this test to pass 

```
val s = """{ "a": 0.0 }"""
JsonAssert.assertJsonEquals(s, s)
```

but it returns:  

```
JSON documents are different:
Different value found in node "a", expected: <0.0> but was: <0>.

net.javacrumbs.jsonunit.core.internal.BasicExceptionFactory$BasicJsonAssertError: JSON documents are different:
Different value found in node "a", expected: <0.0> but was: <0>.
```

**Comment by lukas-krecan** [comment]
https://github.com/lukas-krecan/JsonUnit/#numbers

**Comment by Adven27** [comment]
I'm using jackson2. 
Looks like it's the same issue as #229 and I can workaround it by setting tolerance `JsonAssert.assertJsonEquals(expected, actual, configuration.withTolerance(0.0))` but it's a little not obvious and confusing.

**Comment by lukas-krecan** [comment]
Hi, thanks for feedback. That's strange, the same [test](https://github.com/lukas-krecan/JsonUnit/commit/bd659474581cc0df218f4088a8993ac262d33926) [passes](https://github.com/lukas-krecan/JsonUnit/actions/runs/1813938605) to me. Any idea what could be different in your environment (Jackson plugins, special settings, ...)?

**Comment by Adven27** [comment]
Yes, you're right there is a difference.  I have [USE_BIG_DECIMAL_FOR_FLOATS](https://fasterxml.github.io/jackson-databind/javadoc/2.6/com/fasterxml/jackson/databind/DeserializationFeature.html#USE_BIG_DECIMAL_FOR_FLOATS) setting enabled. 
Else we may have false-positive assertions because of rounding  like this:

```
        val s1 = """{ "a": 1.000000000000000001 }"""
        val s2 = """{ "a": 1.0000000000000001 }"""
        // will pass
        assertThatJson(s1).isEqualTo(s2)
```

I don't whether this is an issue of json-unit or not, but for me, as a library user, it looks like a trade-off between false-negative and false-positive checks... =)

**Comment by lukas-krecan** [comment]
Hi, I assume you are using `Jackson2ObjectMapperProvider` for the configuration. If that's so, you have to specify `USE_BIG_DECIMAL_FOR_FLOATS` for both lenient and normal parser (one is used for parsing the actual value, the other for parsing expected value. See [431d07f](https://github.com/lukas-krecan/JsonUnit/commit/431d07f272948470c52d8741e86375772c91770c))
